pylint/lint/pylinter.py

```diff
@@ -974,10 +971,6 @@ def prepare_checkers(self):
             messages = {msg for msg in checker.msgs if self.is_message_enabled(msg)}
             if messages or any(self.report_is_enabled(r[0]) for r in checker.reports):
                 needed_checkers.append(checker)
-        # Sort checkers by priority
-        needed_checkers = sorted(
-            needed_checkers, key=operator.attrgetter("priority"), reverse=True
-        )
```

Comment:
Is anyone aware of pylint ever having a `failfast` mode? That's all I can think of for the usefulness of sorting checkers. If you want the more important checkers to run first when failing fast.

Reply:
I don't think we have a failfast mode but you're right it's the probably a use case. We could want to order by warnings significance instead of checkers later (i.e. to raise most significant messages first) see #5604

Reply:
We don't have a failfast mode, but this wasn't really documented or implemented, so we can always put one back if we want it and document it.